develop: TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu fail

### Describe the bug:
Windows10 and a linux Mint 20 VM are both failing with the following error
`--- FAIL: TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu (0.78s)
    canvas_other_test.go:43: 
        	Error Trace:	canvas_other_test.go:43
        	Error:      	Expected nil, but got: &glfw.menuBarItem{Base:widget.Base{hidden:false, pos:fyne.Position{X:0, Y:0}, size:fyne.Size{Width:43, Height:29}, propertyLock:sync.RWMutex{w:sync.Mutex{state:0, sema:0x0}, writerSem:0x0, readerSem:0x0, readerCount:0, readerWait:0}}, Menu:(*fyne.Menu)(0xc00020c120), Parent:(*glfw.MenuBar)(0xc00023e000), active:true, child:(*widget.Menu)(0xc000204480), hovered:false}
        	Test:       	TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu
        	Messages:   	activating the menu changes focus handler but does not remove focus from content
    canvas_other_test.go:49: 
        	Error Trace:	canvas_other_test.go:49
        	Error:      	Expected nil, but got: &glfw.menuBarItem{Base:widget.Base{hidden:false, pos:fyne.Position{X:0, Y:0}, size:fyne.Size{Width:43, Height:29}, propertyLock:sync.RWMutex{w:sync.Mutex{state:0, sema:0x0}, writerSem:0x0, readerSem:0x0, readerCount:0, readerWait:0}}, Menu:(*fyne.Menu)(0xc00020c120), Parent:(*glfw.MenuBar)(0xc00023e000), active:true, child:(*widget.Menu)(0xc000204480), hovered:false}
        	Test:       	TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu
        	Messages:   	changing focus with active menu does not affect content focus`
`
It started with the commit
commit 72a7be1565da85fafb9cbefeff4d8ba8a889246f
Author: Tilo Prütz <tilo@pruetz.net>
Date:   Mon Jan 11 21:52:51 2021 +0100

    fix menu bar keyboard control tests
    
    With the changed test helper the expected markup has
    to be in files.`

### To Reproduce:
go test ./... -count=1

### Device (please complete the following information):
 - **OS:** Linux and Windows
 - **Version:** Mint 20 and Windows 10
 - **Go version:** 1.15.6
 - **Fyne version:** develop HEAD

--- a/internal/driver/glfw/canvas_other_test.go
+++ b/internal/driver/glfw/canvas_other_test.go
@@ -19,6 +19,7 @@ func TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu(t *testing.T)
 	w.SetMainMenu(
 		fyne.NewMainMenu(
 			fyne.NewMenu("test", fyne.NewMenuItem("item", func() {})),
+			fyne.NewMenu("other", fyne.NewMenuItem("item", func() {})),
 		),
 	)
 	c := w.Canvas().(*glCanvas)
@@ -38,22 +39,20 @@ func TestGlCanvas_FocusHandlingWhenActivatingOrDeactivatingTheMenu(t *testing.T)
 
 	m.Items[0].(*menuBarItem).Tapped(&fyne.PointEvent{})
 	assert.True(t, m.IsActive())
-	ctxt := "activating the menu changes focus handler but does not remove focus from content"
+	ctxt := "activating the menu changes focus handler and focuses the menu bar item but does not remove focus from content"
 	assert.True(t, ce2.focused, ctxt)
-	assert.Nil(t, c.Focused(), ctxt)
+	assert.Equal(t, m.Items[0], c.Focused(), ctxt)
 
 	c.FocusNext()
-	// TODO: changes menu focus as soon as menu has focus support
 	ctxt = "changing focus with active menu does not affect content focus"
 	assert.True(t, ce2.focused, ctxt)
-	assert.Nil(t, c.Focused(), ctxt)
+	assert.Equal(t, m.Items[1], c.Focused(), ctxt)
 
 	m.Items[0].(*menuBarItem).Tapped(&fyne.PointEvent{})
 	assert.False(t, m.IsActive())
-	// TODO: does not remove focus from focused menu item as soon as menu has focus support
 	ctxt = "deactivating the menu restores focus handler from content"
 	assert.True(t, ce2.focused, ctxt)
-	assert.Equal(t, ce2, c.Focused())
+	assert.Equal(t, ce2, c.Focused(), ctxt)
 
 	c.FocusPrevious()
 	assert.Equal(t, ce1, c.Focused(), ctxt)

